breadbox/breadbox/crud/dimension_types.py

```diff
@@ -277,14 +277,15 @@ def get_dimension_type_metadata_col(
         return {}
 
     values_by_id_tuples = (
-        db.query(TabularCell)
-        .join(TabularColumn)
+        db.query(TabularColumn)
         .filter(
             and_(
                 TabularColumn.dataset_id == dimension_type.dataset_id,
                 TabularColumn.given_id == col_name,
             )
         )
+        .join(TabularCell)
```

Comment:
just curious: is the move of this join down a few lines, a cosmetic change or a performance change?

Reply:
My thinking was that filtering before doing the join would make things faster because there's a much smaller total number of records involved in the join. For example, if there are 20 columns of metadata and 2000 rows, but we only care about the "label" column, this shrinks your join (cross product) to 2000x1=2000 instead of 2000*20=40000.

In this case, it probably doesn't make a meaningful impact though, because I don't think the join is actually the slow part 🤷 I did a little profiling of this later and realized that the `fetch_all` operation (loading the query results) was taking something like 98% of the time this query takes to execute 😓